packages/iov-bcp-types/src/transactions.ts

```diff
@@ -109,15 +112,24 @@ export interface SwapTimeoutTx extends BaseTx {
 export interface RegisterBlockchainTx extends BaseTx {
   readonly kind: TransactionKind.RegisterBlockchain;
   /** the ID of the blockchain to be registered */
-  readonly blockchainId: ChainId;
+  readonly blockchainId: BnsBlockchainId;
+  /** as defined in https://github.com/iov-one/bns-spec/blob/master/docs/data/ObjectDefinitions.rst#chain */
+  readonly chain: {
```

Comment:
We don't have to follow the encoded format here, more what makes most sense on the public side. `codecName` and `codecConfig` are stored under `iov` in the protobuf object.

Maybe `chainId` could go to the top level.
I'm just wondering how much we need to separate them how much we can hide complexity client side.

Reply:
Right. The problem with moving `chainId` up to top level is the conflict with BaseTx.chainId. The only potential I see here is removing `readonly blockchainId: BnsBlockchainId;` as we can set this using `chain.chainId`